### ISSUE TITLE
[Action] Support "no text" display - make `label` required, make `text` optional

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#10190]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Currently, an Action has a required `text` property, and a `label` property that will fallback to the `text` property value if not provided. 

However, it can be valid to not display text within the Action. Currently - to do this you need to pass `text=""` and it creates undesired visuals: 

<img width="552" alt="Screenshot 2024-08-29 at 4 29 02 PM" src="https://github.com/user-attachments/assets/34ffcce8-dacc-45c4-86e4-aaca4d9bb36e">

### Acceptance Criteria

The user should be able to display an action without text - and have it remain accessible through use of a `label` property.

- `label` is made required
- `text` is made optional
- Deprecate `text-enabled` (?? - we could just display it based on presence of text - unless we need this to persist when text is displayed conditionally based on `expanded` behavior of parent Action Bar / Group ??)

### Which Component

Action, Action Bar, Action Group, Action Pad

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (design)

**Comment by driskull** [comment]
>  unless we need this to persist when text is displayed conditionally based on expanded behavior of parent Action Bar / Group ??)

Yes, this is so that users don't have to add/remove text in order to toggle it being displayed.

**Comment by macandcheese** [comment]
Yeah that makes sense. Might be work using `collapse-disabled` or something if we are making other breaking changes to make the use-case for standalone Actions simpler.

**Comment by DitwanP** [comment]
Spike to see if this is still needed.
